Mathlib/CategoryTheory/Presentable/Limits.lean

```diff
@@ -0,0 +1,189 @@
+  dsimp at h₁ h₂ ⊢
+  rw [h₁, hz, FunctorToTypes.naturality, h₂, ← FunctorToTypes.map_comp_apply, cX.w]
+
+lemma injective (j : J) (x₁ x₂ : c.pt.obj (X.obj j))
+    (h : c.pt.map (cX.ι.app j) x₁ = c.pt.map (cX.ι.app j) x₂) :
+    ∃ (j' : J) (α : j ⟶ j'),
+    c.pt.map (X.map α) x₁ = c.pt.map (X.map α) x₂ := by
+  have := isFiltered_of_isCardinalDirected J κ
+  let y₁ := Types.isLimitEquivSections (hc (X.obj j)) x₁
+  let y₂ := Types.isLimitEquivSections (hc (X.obj j)) x₂
+  have hy₁ : (Types.isLimitEquivSections (hc (X.obj j))).symm y₁ = x₁ := by simp [y₁]
+  have hy₂ : (Types.isLimitEquivSections (hc (X.obj j))).symm y₂ = x₂ := by simp [y₂]
+  have H (k : K) := (Types.FilteredColimit.isColimit_eq_iff' (ht := hF k)
+    (x := y₁.1 k) (y := y₂.1 k)).1 (by
+      simp only [y₁, y₂, Types.isLimitEquivSections_apply]
+      dsimp
+      simp only [← FunctorToTypes.naturality, h])
+  let j₁ (k : K) : J := (H k).choose
+  let f (k : K) : j ⟶ j₁ k := (H k).choose_spec.choose
+  have hf (k : K) : (F.obj k).map (X.map (f k)) (y₁.1 k) =
+      (F.obj k).map (X.map (f k)) (y₂.1 k) :=
+    (H k).choose_spec.choose_spec
+  have hK' := hasCardinalLT_of_hasCardinalLT_arrow hK
+  let ψ (k : K) : j ⟶ IsCardinalFiltered.max j₁ hK' :=
+    f k ≫ IsCardinalFiltered.toMax j₁ hK' k
+  refine ⟨IsCardinalFiltered.coeq ψ hK', IsCardinalFiltered.toCoeq ψ hK', ?_⟩
+  apply (Types.isLimitEquivSections (hc _)).injective
+  ext k
+  simp only [Types.isLimitEquivSections_apply, ← hy₁, ← hy₂]
+  have h₁ := Types.isLimitEquivSections_symm_apply (hc (X.obj j)) y₁ k
+  have h₂ := Types.isLimitEquivSections_symm_apply (hc (X.obj j)) y₂ k
+  dsimp at h₁ h₂ ⊢
+  simp only [FunctorToTypes.naturality, h₁, h₂,
+    ← IsCardinalFiltered.coeq_condition ψ hK' k,
+    map_comp, FunctorToTypes.map_comp_apply, ψ, hf]
+
+end isColimitMapCocone
+
+/-- Auxiliary definition for `isCardinalAccessible_of_isLimit`. -/
+noncomputable def isColimitMapCocone : IsColimit (c.pt.mapCocone cX) := by
+  have := isFiltered_of_isCardinalDirected J κ
+  apply Types.FilteredColimit.isColimitOf'
+  · exact isColimitMapCocone.surjective c hc κ hK cX hF
+  · exact isColimitMapCocone.injective c hc κ hK cX hF
+
+end
+
+end Limits
+
+end Accessible
+
+lemma isCardinalAccessible_of_isLimit {K : Type u'} [Category.{v'} K] {F : K ⥤ C ⥤ Type w'}
+    (c : Cone F) (hc : IsLimit c) (κ : Cardinal.{w}) [Fact κ.IsRegular]
+    [HasLimitsOfShape K (Type w')] (hK : HasCardinalLT (Arrow K) κ)
+    [∀ k, (F.obj k).IsCardinalAccessible κ] :
+    c.pt.IsCardinalAccessible κ where
+  preservesColimitOfShape {J _ _} := ⟨fun {X} ↦ ⟨fun {cX} hcX ↦ by
+    have := fun k ↦ preservesColimitsOfShape_of_isCardinalAccessible (F.obj k) κ J
+    exact ⟨Accessible.Limits.isColimitMapCocone c
+      (fun Y ↦ isLimitOfPreserves ((evaluation C (Type w')).obj Y) hc) κ hK cX
+      (fun k ↦ isColimitOfPreserves (F.obj k) hcX)⟩⟩⟩
+
+end Functor
+
+/-- This is `isCardinalPresentable_of_isColimit` in the particular case `w = v`. -/
```

Comment:
Small nitpick is that I would prefer if `isCardinalPresentable_of_isColimit'` was stated using `[HasLimitsOfShape Kᵒᵖ (Type v)]` as a hypothesis instead of fixing the universe parameters. This makes it more similar to `isCardinalAccessible_of_isLimit` and in the fixed universe parameter case typeclass inference should be able to figure this out anyways. But doesnt matter much in practice I guess, so up to you!

Reply:
Thanks for the suggestion!